### ISSUE TITLE
Button is inconsistent when dispatching messages

### ISSUE THREAD

**Issue by saqimtiaz** [Jermolene/TiddlyWiki5#2908]
Currently there is significant inconsistency in the information included in a message dispatched by the Button widget, when comparing "tm-navigate" messages to all other messages.

Granted tm-navigate messages have some attributes that are specific to those messages, such as navigateTo. But other attributes such as navigateFromNode and navigateFromClientRect should also be set for other messages that are dispatched, but are not.
See https://github.com/Jermolene/TiddlyWiki5/blob/master/core/modules/widgets/button.js#L134

One use case is determining from the dispatched message, which story a button was clicked in. While this is possible for buttons for navigation via the navigateFromNode reference to the button widget, it is not possible with the edit button for tiddlers as there is no reference to the widget that sent the message.

**Comment by tobibeer** [comment]
I agree that the message model would totally benefit from a more harmonized set of details being passed down.

I suppose at some point the focus was more on actionwidgets and less on the message side of things.

**Comment by pmario** [comment]
>I suppose at some point the focus was more on actionwidgets and less on the message side of things.

I think most of the stuff was created for a single usecas at the beginning. .. tm-navigate needed more info, so it got it. 

Refactoring all the other widgets, that _could have_ benefitted is just a lot of work ... and nobody asked for it  ... yet ;)

**Comment by tobibeer** [comment]
In short, someone ought to make a PR for the respective message(s) to avail those additional details.

**Comment by pmario** [comment]
@saqimtiaz -- Is this issue still relevant? If no please close.
